SimDataFormats/TrackingAnalysis/interface/TrackingParticle.h

```diff
@@ -173,12 +182,28 @@ class TrackingParticle {
     const TrackingVertex& r = (*parentVertex_);
     return r.position().Y();
   }
-  // @brief z coordinate of parent vertex position
+
+  /// @brief z coordinate of parent vertex position
   double vz() const {
     const TrackingVertex& r = (*parentVertex_);
     return r.position().Z();
   }
 
+  /// @brief dxy parameter. (This is the transverse impact parameter w.r.t. to (0,0,0) ONLY if refPoint is close to (0,0,0)).
+  double dxy() const { return (-vx() * py() + vy() * px()) / pt(); }
```

Comment:
The comment is misleading because the `vx()` etc are the production point of the simulated particle. If the point is to get impact parameters with respect to the beamline or `(0,0,0)` I'd suggest to look [`ParametersDefinerForTP`](https://github.com/cms-sw/cmssw/blob/master/SimTracker/TrackAssociation/interface/ParametersDefinerForTP.h) that e.g. tracking validation uses for that purpose.